### PR TITLE
Added an introduction page

### LINKED ISSUE #70: Better docs for new users
By @alexryderwhytewaters:

I just came here to suggest that you allow cache path to be changed via the constructor, but it looks like that has already been done. However the lastest composer package doesn't include that change yet. 

While I am at it, I will just mention a couple of other things (think of them as feedback, not good, not bad, just feedback from a first time user)

Your site has an example of how to do tagging and more complex stuff, but the only place I could find a simple example how how to create a pool is in github of the actual adapters themselves (maybe i missed something, but it wasn't easy to find). It would be nice if you can just add a 5 line example to the main site, showing accessing of a pool eg. and getting and item then saving it. 

Next issue i ran into is, i am caching webpages based on url, so the first thing i did is
$cachePool->getItem($_SERVER['REQUEST_URI']);

which failed, because you can't include certain characters for future features. So I am now using it like this
$cachePool->getItem(md5($_SERVER['REQUEST_URI']));

To me, the cache key, should not have reserved characters, maybe a separate parameter to provide extra features, but that's up to you, just feedback.

Next I went to add a tag, same thing limited to what i can use as characters, : not allowed
$item->set($html)->addTag("operator:{$operator->operator_rn}");

So now I am using underscore, to me it would seem more logical to allow any char and just encode the data, eg url_encode
$item->set($html)->addTag("operator_{$operator->operator_rn}");

Both issues, I struck are easy to work around, I only mentioned them as they were first impressions,
and I will have forgotten about them in a few weeks time

Your cache package seems to be the only filesystem one that handles tagging that i have found so far, so thank you for providing that.

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix https://github.com/php-cache/issues/issues/70

### CODE REVIEW

File: docs/introduction.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# PSR-6 Cache Introduction
+ 
+In the late fall 2015 the [PHP-FIG](http://www.php-fig.org/) released a specification
+for caching. It is called the [PSR-6](http://www.php-fig.org/psr/psr-6/). Like all the
```
Comment (cryptiklemur) [inline docs/introduction.md]: `It is called [PSR-6]`

File: docs/introduction.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# PSR-6 Cache Introduction
+ 
+In the late fall 2015 the [PHP-FIG](http://www.php-fig.org/) released a specification
+for caching. It is called the [PSR-6](http://www.php-fig.org/psr/psr-6/). Like all the 
+PSRs it is just a recommendation for improve interoperability.
```
Comment (cryptiklemur) [inline docs/introduction.md]: `Like all the PSRs, it is just`

File: docs/introduction.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+$pool->deleteItem('cache_key');
+$pool->hasItem('cache_key'); // False
+```
+ 
+ 
+*Note: A cache item can not be transferred from one pool to another.*
+
+## Cache keys
+
+According to the PSR-6 the following characters could be used in a valid cache key;
```
Comment (cryptiklemur) [inline docs/introduction.md]: `According to PSR-6`

File: docs/introduction.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+To make sure you do not use an invalid character by mistake, you should hash your keys. 
+ 
+```php
+$cacheKey = sha1($_SERVER['REQUEST_URI']);
+```
+
+## Choosing a pool
+
+What cache pool you  want to use is up to you. As you can in the
```
Comment (cryptiklemur) [inline docs/introduction.md]: `As you can see in the`

File: docs/introduction.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+```php
+$client = new \Memcached();
+$client->addServer('localhost', 11211);
+$pool = new MemcachedCachePool($client);
+```
+
+
+## Features
+
+The PHP-Cache organization has build some more features on top of PSR-6. Look at our
```
Comment (cryptiklemur) [inline docs/introduction.md]: `organization has built some`

**Comment by ghost** [comment]
Yeah it looks good to me, just out of curiosity what is an ApcuCachePool? I have heard apc before but not Apcu

**Comment by cryptiklemur** [comment]
APCu is userland APC Cache. In 5.6 (IIRC), they added the APC opcode cache to PHP's core, so it wasn't needed anymore. Gave way to the new APCu library

**Comment by Nyholm** [comment]
Thank you for the feedback

**Pull request merged**